### PR TITLE
Grafana HA, step1

### PR DESCRIPTION
**Pull request by umezawatakeshi**
add MySQL for Grafana to use it later.

Grafana's configuration will be changed in https://github.com/cybozu-go/neco-apps/pull/2006

part of https://github.com/cybozu-go/neco/issues/1805

Signed-off-by: UMEZAWA Takeshi <takeshi-umezawa@cybozu.co.jp>

### CODE REVIEW

File: monitoring/base/grafana-operator/mysql.yaml [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+            app.kubernetes.io/name: mysql
+            app.kubernetes.io/instance: grafana-mysql # must be same as MySQLCluster's name
+        maxSkew: 1
+        topologyKey: topology.kubernetes.io/zone
+        whenUnsatisfiable: DoNotSchedule
+  volumeClaimTemplates:
+  - metadata:
+      name: mysql-data
+    spec:
+      storageClassName: topolvm-provisioner
```
Comment (morimoto-cybozu) [inline monitoring/base/grafana-operator/mysql.yaml:25]: `topolvm-flash1-xfs` or `topolvm-flash1-ext4` is recommended.

File: monitoring/base/grafana-operator/mysql.yaml [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+kind: MySQLCluster
+metadata:
+  namespace: monitoring
+  name: grafana-mysql
+spec:
+  replicas: 3
+  podTemplate:
+    spec:
+      containers:
+      - name: mysqld
```
Comment (morimoto-cybozu) [inline monitoring/base/grafana-operator/mysql.yaml:11]: Neco users wiki recommends to configure `resources.limits` for the MySQLd container.

File: monitoring/base/grafana-operator/mysql.yaml [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+      - labelSelector:
+          matchLabels:
+            app.kubernetes.io/name: mysql
+            app.kubernetes.io/instance: grafana-mysql # must be same as MySQLCluster's name
+        maxSkew: 1
+        topologyKey: topology.kubernetes.io/zone
+        whenUnsatisfiable: DoNotSchedule
+  volumeClaimTemplates:
+  - metadata:
+      name: mysql-data
```
Comment (morimoto-cybozu) [inline monitoring/base/grafana-operator/mysql.yaml:23]: Neco users wiki recommends to enable auto-resizing.

**Review by zoetrope — DISMISSED** [review]
LGTM

File: monitoring/base/victoriametrics/rules/monitoring-alertrule.yaml [hunk 4]
```diff
@@ -176,3 +176,21 @@ spec:
           annotations:
             summary: grafana is down.
             runbook: TBD
+        - alert: GrafanaMySQLDown
+          expr: |
+            absent(up{job="moco-mysql",name="grafana-mysql",namespace="monitoring"} == 1) or count(up{job="moco-mysql",name="grafana-mysql",namespace="monitoring"} == 1) < 2
```
Comment (morimoto-cybozu) [inline monitoring/base/victoriametrics/rules/monitoring-alertrule.yaml:181]: `moco-controller` provides the [metrics about MySQL clusters](https://github.com/cybozu-go/moco/blob/main/docs/metrics.md).
I'm not sure whether we need to monitor MySQLs and agents directly.

**Review by morimoto-cybozu — CHANGES_REQUESTED** [review]

File: monitoring/base/victoriametrics/rules/monitoring-alertrule.yaml (hunk 4 above)
Comment (ymmt2005) [reply monitoring/base/victoriametrics/rules/monitoring-alertrule.yaml:181]: We can use `moco_cluster_healthy` and alert if the cluster is not healthy for a long time.

File: monitoring/base/grafana-operator/mysql.yaml (hunk 1 above)
Comment (umezawatakeshi) [reply monitoring/base/grafana-operator/mysql.yaml:25]: which is recommended?

**Review by morimoto-cybozu — APPROVED** [review]
LGTM

**Pull request merged**